_posts/2020-02-06-how-to-configure-a-production-grade-ci-cd-setup-for-apps-and-infrastructure-code.adoc

```diff
@@ -0,0 +1,2045 @@
+---
+title: How to configure a production-grade CI/CD workflows for application and infrastructure code
+categories: Foundations
+image: TODO
+excerpt: Learn about CI/CD workflows for application and infrastructure code, including the differences between the two, different CI servers, threat models, and more.
+tags: ["aws", "terraform", "cicd"]
+cloud: ["aws"]
+---
+:page-type: guide
+:page-layout: post
+
+:toc:
+:toc-placement!:
+
+// GitHub specific settings. See https://gist.github.com/dcode/0cfbf2699a1fe9b46ff04c41721dda74 for details.
+ifdef::env-github[]
+:tip-caption: :bulb:
+:note-caption: :information_source:
+:important-caption: :heavy_exclamation_mark:
+:caution-caption: :fire:
+:warning-caption: :warning:
+toc::[]
+endif::[]
+
+== Intro
+
+This guide is a comprehensive overview of how to design, configure, and implement a Continuous Integration and
+Continuous Delivery pipeline for your application and infrastructure code.
+
+=== What is Continuous Integration and Continuous Delivery?
+
+Continuous Integration and Continuous Delivery (also widely known as CI/CD) is a software development practice that
+involves developers merging their work together and deploying it to production on a regular basis (oftentimes as
+frequent as multiple times per day). The goal of a Continuous Integration process is to integrate the various feature
+work that the developers are individually doing often enough such that you can identify problems with the design earlier
+in the process, allowing you to improve the design incrementally. Similarly, by deploying the software more frequently
+to production, the Continuous Delivery process enables you to keep software packages small enough to reduce the risk and
+impact of each deployment.
+
+While CI/CD for application code is well understood in the software industry, CI/CD for infrastructure code is a
+relatively new concept. This guide focuses on providing an overview of the background info, design, and implementation
+of a production-ready CI/CD pipeline for infrastructure code.
+
+
+=== What you'll learn in this guide
```

Comment:
Might want to mention this guide will show you how to set up a secure CI / CD pipeline using your favorite CI / CD platform (e.g., Jenkins, Circle, GitLab, etc)

Reply:
Good idea. Added to top paragraph: 54b5aac311d78d27b35025469ebf47213e9a8b48